sql/plsql/PlSqlParser.g4

```diff
@@ -1814,6 +1814,23 @@ create_materialized_view
         ';'
     ;
 
+scoped_table_ref_constraint
+    : SCOPE FOR '(' ref_column_or_attribute ')' IS 
+        (schema_name '.')? scope_table_name_or_c_alias
+    ;
+
+ref_column_or_attribute
+    : identifier
+    ;
+
+scope_table_name_or_c_alias
+    : identifier
+    ;
```

Comment:
Do we have strong need of introducing new rules that only have `identifier` inside? Could `identifier` be inlined in the respective use-sites?

```suggestion
scoped_table_ref_constraint
    : SCOPE FOR '(' identifier ')' IS (schema_name '.')? identifier
    ;
```

Also, [rule element labels](https://github.com/antlr/antlr4/blob/master/doc/parser-rules.md#rule-element-labels) can be used if necessary:

```sql
scoped_table_ref_constraint
    : SCOPE FOR '(' ref_column_or_attribute=identifier ')' IS (schema_name '.')? scope_table_name_or_c_alias=identifier
    ;
```

Reply:
Thanks for pointing out the labeling feature, where I was not aware of. It simplifies the the resulting parser implementation while keeping the information within the grammar.